src/ir/module-utils.h

```diff
@@ -416,20 +416,27 @@ collectSignatures(Module& wasm,
       TypeCounter(Counts& counts) : counts(counts) {}
 
       void visitExpression(Expression* curr) {
-        if (curr->is<RefNull>()) {
-          auto heapType = curr->type.getHeapType();
-          if (heapType.isSignature()) {
-            counts[heapType.getSignature()]++;
-          }
+        if (curr->is<RefNull>() || curr->is<RefFunc>()) {
```

Comment:
It doesn't look like `ref.func` has a type annotation: https://webassembly.github.io/reference-types/core/binary/instructions.html#reference-instructions/

Reply:
I may be confused. While it doesn't have a type in the binary directly, it does potentially define a new type - say, a nullable reference to a function of type X, which differs from the type X itself. That new type may appear in no other places. Is that not a problem?

Reply:
Seems like it shouldn't be, as you say. Fuzzer agrees. Fixed!

Reply:
Yeah, it's kind of weird that there are runtime types that do not appear in the type section, now that I think about it. Glad the fuzzer agrees that this is correct, though!